src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/StateSynchronizationMenuItems.cs

```diff
@@ -87,53 +87,93 @@ public static void ClearAllAssetCaches()
+            public BuildTarget BuildTarget;
+        }
+
         [MenuItem("Spectator View/Generate Asset Bundles", priority = 102)]
         public static void GenerateAssetBundles()
         {
-            string iOSDirectory = Application.dataPath + $"/{AssetCache.assetCacheDirectory}/{ResourcesDirectoryName}/{nameof(AssetBundlePlatform.iOS)}";
-            string androidDirectory = Application.dataPath + $"/{AssetCache.assetCacheDirectory}/{ResourcesDirectoryName}/{nameof(AssetBundlePlatform.Android)}";
-            string wsaDirectory = Application.dataPath + $"/{AssetCache.assetCacheDirectory}/{ResourcesDirectoryName}/{nameof(AssetBundlePlatform.WSA)}";
-
-            Directory.CreateDirectory(iOSDirectory.Replace('/', Path.DirectorySeparatorChar));
-            Directory.CreateDirectory(androidDirectory.Replace('/', Path.DirectorySeparatorChar));
-            Directory.CreateDirectory(wsaDirectory.Replace('/', Path.DirectorySeparatorChar));
-
-            List<string> builtDirectories = new List<string>();
+            var bundlesToCreate = new List<AssetBundleInfo>();
 
 #if UNITY_EDITOR_OSX
-            EditorUserBuildSettings.SwitchActiveBuildTarget(BuildTargetGroup.iOS, BuildTarget.iOS);
-            BuildPipeline.BuildAssetBundles(iOSDirectory, BuildAssetBundleOptions.None, BuildTarget.iOS);
-            builtDirectories.Add(iOSDirectory);
-#elif UNITY_EDITOR_WIN
-            EditorUserBuildSettings.SwitchActiveBuildTarget(BuildTargetGroup.Android, BuildTarget.Android);
-            BuildPipeline.BuildAssetBundles(androidDirectory, BuildAssetBundleOptions.None, BuildTarget.Android);
-
-            EditorUserBuildSettings.SwitchActiveBuildTarget(BuildTargetGroup.WSA, BuildTarget.WSAPlayer);
-            BuildPipeline.BuildAssetBundles(wsaDirectory, BuildAssetBundleOptions.None, BuildTarget.WSAPlayer);
-
-            builtDirectories.Add(androidDirectory);
-            builtDirectories.Add(wsaDirectory);
+            bundlesToCreate.Add(new AssetBundleInfo
+            {
+                AssetBundlePlatform = AssetBundlePlatform.iOS,
+                BuildTargetGroup = BuildTargetGroup.iOS,
+                BuildTarget = BuildTarget.iOS,
+            });
+#else
+            bundlesToCreate.Add(new AssetBundleInfo
+            {
+                AssetBundlePlatform = AssetBundlePlatform.Android,
+                BuildTargetGroup = BuildTargetGroup.Android,
+                BuildTarget = BuildTarget.Android,
+            });
+
+            bundlesToCreate.Add(new AssetBundleInfo
+            {
+                AssetBundlePlatform = AssetBundlePlatform.WSA,
+                BuildTargetGroup = BuildTargetGroup.WSA,
+                BuildTarget = BuildTarget.WSAPlayer,
+            });
 #endif
 
-            foreach (var directory in builtDirectories)
+            // TODO: save current build target and restore it at the end... perhaps build the current target bundle before switching (or maybe at the end).
+
```

Comment:
we should definitely build the current target before switching, for most of these scoped apps its easy to switch players. For actual end user applications it can take 15-30 minutes just to switch players.